### ISSUE TITLE
AttributeError on Blender 3.1

### ISSUE THREAD

**Issue by sionleroux** [JacquesLucke/Blender-Camera-Addon#14]
If I try to enable the add-on in Blender 3.1.2 then it immediately spits out out an error while trying to register:
![image](https://user-images.githubusercontent.com/840466/169855452-8a1f5e10-6d34-4291-bcf3-c5c50bb7d01c.png)

```
AttributeError: module 'bpy.utils' has no attribute register_module
```

Could this be an API compatibility issue with newer versions of Blender?

Oh, meanwhile I see the docs say that `register_module` [has been removed](https://wiki.blender.org/wiki/Reference/Release_Notes/2.80/Python_API/Addons) but the this is for version 2.80 and the compatibility version in __init__.py also says 2.80, am I looking at a wrong branch?

**Comment by sionleroux** [comment]
Okay I couldn't find mentions of `register_module` in the code on GitHub so I think I just had an older version, so I deleted everything, downloaded it again from the master branch, restart Blender and now it works fine, sorry for the noise.

**Comment by sionleroux** [comment]
Ok what about this?

![image](https://user-images.githubusercontent.com/840466/169861734-8b4cb70b-17e5-413a-94ba-b62e738b3443.png)

'SNIPER_OT_select_target' object has no attribute 'currentIndex'
location: <unknown location>:-1

Happens when I click "Here" button (the name of the text) or either of the arrows to the left/right of it:
![screenshot](https://user-images.githubusercontent.com/840466/169862191-d9a4b30b-61aa-4d86-a95e-83dbc72314e7.png)

**Comment by infamanious** [comment]
The same error occurs on blender 3.3.  Might be a deprecation issue im not sure, but i get TypeError: list indices must be integers or slices, not str. Because of this error i can't access the properties for adjusting wiggle wiggle wiggle and whatnot.

**Comment by Spikerama** [comment]
Hi guys, I'm wondering if you can help me. 

How are you installing Sniper into newer versions of Blender? I've tried it on every version from 2.79 to 3.3. I can only get it to work on 2,79. I think mainly because on 2.79 the option to chose File tab on the addons preferences is there. That way I can allow "Auto Run of Pi Scripts." 

This tab isn't in the future versions of Blender so maybe that's the issue because Sniper doesn't appear in the Tools Shelf on the later installs  I do. 

Anyway I'd love to get to the bottom of it, so any advice would be very much appreciated. 

I used the Sniper 1.3.2 version.

**Comment by schroef** [comment]
> Hi guys, I'm wondering if you can help me.
> 
> How are you installing Sniper into newer versions of Blender? I've tried it on every version from 2.79 to 3.3. I can only get it to work on 2,79. I think mainly because on 2.79 the option to chose File tab on the addons preferences is there. That way I can allow "Auto Run of Pi Scripts."
> 
> This tab isn't in the future versions of Blender so maybe that's the issue because Sniper doesn't appear in the Tools Shelf on the later installs I do.
> 
> Anyway I'd love to get to the bottom of it, so any advice would be very much appreciated.
> 
> I used the Sniper 1.3.2 version.

Try this version > https://github.com/IIK3D/Blender-Camera-Addon/tree/Update_2.80x

**Comment by Spikerama** [comment]
> Try this version > https://github.com/IIK3D/Blender-Camera-Addon/tree/Update_2.80x

Hey thanks so much @schroef for the speedy response. I will give it a go.

**Comment by Spikerama** [comment]
@schroef Yes it works a treat mate. Thanks again. The only thing I can't get to work is the Manipulate Timing button which is inactive. But I guess it's the same as opening the dope sheet?

**Comment by schroef** [comment]
@Spikerama 

Try a google search. I did that last week with this addon to remind me how it works. This repo shows zero feedback what it does and how it works. I believe the action editor should open, that happened for me when i pressed it

Have a look at this link

https://www.google.com/url?sa=t&source=web&rct=j&url=https://www.blenderdiplom.com/en/downloads/569-addon-sniper-camera-automation.html&ved=2ahUKEwiEzMDOzOn9AhUNTDABHX-rAXIQFnoECA8QAQ&usg=AOvVaw3OJRM8f9F9rQCwlURFQvGn

**Comment by Spikerama** [comment]
@schroef Thanks mate, yeah I did check that out. There is also a youtube video that was done for the original release which I studied. It just seems that the "Manipulate Timing" button is inactive by default. It might be a version issue as it works on 2.79. 

Anyway it's not a major issue because you can just open the Dope Sheet and fiddle around that way. 

Cheers again.

**Comment by schroef** [comment]
Weird, for me in 2.83 it is active. Do you get any errors in the console?

Only thing it actually does is open a dipsheet window. Which we need to manually remove later

**Comment by schroef** [comment]
Weird, for me in 2.83 it is active. Do you get any errors in the console?

Only thing it actually does is open a dipsheet window. Which we need to manually remove later

Edit
I did some more testing. This addon has some more issues. For me that notification about the need to update never goes away. Some variable is set globally but not properly. I fixed that part.

Now i also see what you mean with the edit timeline. Weird thing is, it's checking if a dipsheet editor is open. But when you have a timeline open, it returns that as a dopesheet which is weird, the areatype should be timline

**Comment by Spikerama** [comment]
Yeah I have the same issue with needing to update to version 2.8 never going away. 

Well you are a smarter one than me being able to delve into the script and root out problems. I have no idea on that side of things. Maybe I could get ChatGTP to have a look, that's about it. Haha. 

In the Addon he has also got these camera actions which seem to be working ok. Explained quite well in this video

https://www.youtube.com/watch?v=wcjJLhdeBz4

**Comment by schroef** [comment]
I did some minor work and know seems to work nicely. I've also added a trigger to close the nee area it adds when you click "manipulate timeline"

**Comment by Spikerama** [comment]
> 

Excellent. Sounds much better. Did you push it to one of the repositories?

**Comment by Spikerama** [comment]
> I did some minor work and know seems to work nicely. I've also added a trigger to close the nee area it adds when you click "manipulate timeline"

Hi again mate. Just wondering if you shared your updates on this? Would love to try. Cheers.
